sudoku_generator.py

```diff
@@ -2,12 +2,26 @@
 
 # one for each character position, space margins around them,
 # and border spots for pipes.
+import random
```

Comment:
Cool, you decided to import a handle for the random package; it also would have been fine in this case to do `from random import randint` , but this works great.